modules/acknowledgements/php/acknowledgements.class.inc

```diff
@@ -93,28 +93,28 @@ class Acknowledgements extends \NDB_Menu_Filter_Form
             if ($values['addPresent'] === 'Yes') {
                 $values['addEndDate'] = null;
             }
-            $affiliations = is_array($values['addAffiliations']) ?
+            $affils  = is_array($values['addAffiliations']) ?
                 implode(', ', $values['addAffiliations']) :
                 $values['addAffiliations'];
-            $degrees      = is_array($values['addDegrees']) ?
+            $degs    = is_array($values['addDegrees']) ?
                 implode(', ', $values['addDegrees']) :
                 $values['addDegrees'];
-            $roles        = is_array($values['addRoles']) ?
+            $roles   = is_array($values['addRoles']) ?
                 implode(', ', $values['addRoles']) :
                 $values['addRoles'];
-            $results      = [
+            $results = [
                 'ordering'      => $values['addOrdering'],
                 'full_name'     => $values['addFullName'],
                 'citation_name' => $values['addCitationName'],
-                'affiliations'  => $affiliations,
-                'degrees'       => $degrees,
-                'roles'         => $roles,
+                'affiliations'  => implode(',', array_filter(explode(',', $affils))),
```

Comment:
why not do the array_filter before the implode on line 96/99/102?

Reply:
I think since the fields are strings and not arrays, it skips lines 96, 99, and 102. I didn't want to get rid of that part though as I don't know if there was some other reason for it. By doing it later it makes sure to do it for all cases without duplicating the codes on 96 and 97, 99 and 100, and 102 and 103
![image](https://github.com/aces/Loris/assets/51128536/94e2127b-badd-4e69-b1b0-b697c40cca68)